src/containerDeployment/dockerUtils.ts

```diff
@@ -242,14 +362,103 @@ export function sanitizeContainerInput(name: string): string {
 //#region Docker Command Implementations
 
 /**
- * Helper function to execute a command in the shell and return the output.
+ * Interface for parameterized commands
+ */
+interface DockerCommand {
+    command: string;
+    args: string[];
+}
+
+/**
+ * Safe command execution helper that uses spawn
+ */
+async function execDockerCommand(cmd: DockerCommand): Promise<string> {
+    return new Promise((resolve, reject) => {
+        const process = spawn(cmd.command, cmd.args, {
+            stdio: ["ignore", "pipe", "pipe"],
+        });
+
+        let stdout = "";
+        let stderr = "";
+
+        process.stdout.on("data", (data) => {
+            stdout += data.toString();
+        });
+
+        process.stderr.on("data", (data) => {
+            stderr += data.toString();
+        });
+
+        process.on("close", (code) => {
+            if (code === 0) {
+                resolve(stdout.trim());
+            } else {
+                const error = new Error(stderr || `Command failed with exit code ${code}`);
+                (error as any).code = code;
+                reject(error);
+            }
+        });
+
+        process.on("error", (error) => {
+            reject(error);
+        });
+    });
+}
+
+/**
+ * Safe PowerShell command execution helper
+ */
+async function execPowerShellCommand(cmd: DockerCommand): Promise<string> {
+    const powerShellExecutable = platform() === "win32" ? "powershell.exe" : "pwsh";
+    return execDockerCommand({
+        command: powerShellExecutable,
+        args: ["-Command", ...cmd.args],
+    });
+}
+
+/**
+ * Safe system command execution helper for platform-specific system operations
+ */
+async function execSystemCommand(cmd: DockerCommand): Promise<string> {

```

Comment:
Does this function need to exist?  Can callers just call `execDockerCommand()` directly?